File: study/lua/src/app/scene/view/heroDetail/HeroDetailBriefModule.lua
--
-- Author: Liangxu
-- Date: 2017-03-01 14:22:05
-- 武将详情 简介模块
local ListViewCellBase = require("app.ui.ListViewCellBase")
local HeroDetailBriefModule = class("HeroDetailBriefModule", ListViewCellBase)
local CSHelper = require("yoka.utils.CSHelper")

function HeroDetailBriefModule:ctor(heroUnitData)
	self._heroUnitData = heroUnitData

	local resource = {
		file = Path.getCSB("HeroDetailDynamicModule", "hero"),
		binding = {

		}
	}
	HeroDetailBriefModule.super.ctor(self, resource)
end

function HeroDetailBriefModule:onCreate()
	local title = self:_createTitle()
	self._listView:pushBackCustomItem(title)

	local des = self:_createDes()
	self._listView:pushBackCustomItem(des)

	self._listView:doLayout()
	local contentSize = self._listView:getInnerContainerSize()
	contentSize.height = contentSize.height + 10
	self._listView:setContentSize(contentSize)
	self:setContentSize(contentSize)
end

function HeroDetailBriefModule:_createTitle()
	local title = CSHelper.loadResourceNode(Path.getCSB("CommonDetailTitleWithBg", "common"))
	title:setFontSize(24)
	title:setTitle(Lang.get("hero_detail_title_brief"))
	local widget = ccui.Widget:create()
	local titleSize = cc.size(402, 36)
	local widgetSize = cc.size(402, 36 + 10)
	widget:setContentSize(widgetSize)
	title:setPosition(titleSize.width / 2, titleSize.height / 2 + 7)
	widget:addChild(title)

	return widget
end

function HeroDetailBriefModule:_createDes()
	local briefDes = self._heroUnitData:getConfig().description
	local color = Colors.BRIGHT_BG_TWO

	local widget = ccui.Widget:create()
	local labelDes = cc.Label:createWithTTF(briefDes, Path.getCommonFont(), 20)
	labelDes:setAnchorPoint(cc.p(0, 1))
	labelDes:setLineHeight(26)
	labelDes:setWidth(350)
	labelDes:setColor(color)

	local height = labelDes:getContentSize().height
	labelDes:setPosition(cc.p(24, height + 15))
	widget:addChild(labelDes)

	local size = cc.size(402, height + 20)
	widget:setContentSize(size)

	return widget
end

return HeroDetailBriefModule

File: study/ts/script/scene/view/heroDetail/HeroDetailBriefModule.ts
import { HeroUnitData } from "../../../data/HeroUnitData";
import { Colors } from "../../../init";
import { Lang } from "../../../lang/Lang";
import { CommonDetailModule } from "../../../ui/component/CommonDetailModule";
import CommonDetailTitleWithBg from "../../../ui/component/CommonDetailTitleWithBg";
import ListViewCellBase from "../../../ui/ListViewCellBase";
import { Path } from "../../../utils/Path";
import UIHelper from "../../../utils/UIHelper";

const { ccclass, property } = cc._decorator;

@ccclass
export default class HeroDetailBriefModule extends ListViewCellBase implements CommonDetailModule {
    @property({
        type: cc.Node,
        visible: true
    })
    _listView: cc.Node = null;

    onCreate() {
        this._commonDetailTitleWithBg = cc.resources.get(Path.getCommonPrefab("CommonDetailTitleWithBg"));
    }

    private _commonDetailTitleWithBg;
    private _heroUnitData: HeroUnitData;
    public setInitData(heroUnitData: HeroUnitData): void {
        this._heroUnitData = heroUnitData;
    }

    public numberOfCell(): number {
        return 2;
    }

    public cellAtIndex(i: number): cc.Node {
        if (i == 0) {
            var title = this._createTitle();
            return title;
        } else {
            var des = this._createDes();
            return des.node;
        }
    }

    public sectionView(): cc.Node {
        return (this._listView.getChildByName("content") as cc.Node);
    }

    public footerHeight(): number {
        return 10;
    }

    _createTitle() {
        var title = (cc.instantiate(cc.resources.get(Path.getCommonPrefab("CommonDetailTitleWithBg"))) as cc.Node).getComponent(CommonDetailTitleWithBg)
        title.setFontSize(24);
        title.setTitle(Lang.get('avatar_detail_brief_title'));
        var widget = new cc.Node();
        var titleSize = cc.size(402, 36);
        var widgetSize = cc.size(402, 36 + 10);
        widget.setContentSize(widgetSize);
        title.node.setPosition(titleSize.width / 2, titleSize.height / 2-20);
        widget.addChild(title.node);
        return widget;
    }
    _createDes() {
        var briefDes = this._heroUnitData.getConfig().description;
        var color = Colors.BRIGHT_BG_TWO;

        var labelDes: cc.Label = UIHelper.createWithTTF(briefDes, Path.getCommonFont(), 20);
        //自动换行

        labelDes.node.setAnchorPoint(new cc.Vec2(0, 1));
        labelDes.verticalAlign = cc.Label.VerticalAlign.TOP;
        labelDes.lineHeight = (26);
        labelDes.node.width = 350;
        labelDes.overflow = cc.Label.Overflow.RESIZE_HEIGHT;
        labelDes.node.color = (color);
        labelDes.node.x = 24;

        labelDes['_updateRenderData']();
        return labelDes;
    }

}